async_asgi_testclient/testing.py

```diff
@@ -78,33 +79,46 @@ def __init__(
         self._lifespan_task = None  # Must keep hard reference to prevent gc
 
     async def __aenter__(self):
-        self._lifespan_task = create_monitored_task(
-            self.application(
-                {"type": "lifespan", "asgi": {"version": "3.0"}},
-                self._lifespan_input_queue.get,
-                self._lifespan_output_queue.put,
-            ),
-            self._lifespan_output_queue.put_nowait,
-        )
+        try:
+            self._lifespan_task = create_monitored_task(
+                self.application(
+                    {"type": "lifespan", "asgi": {"version": "3.0"}},
+                    self._lifespan_input_queue.get,
+                    self._lifespan_output_queue.put,
+                ),
+                self._lifespan_output_queue.put_nowait,
+            )
+            # Make sure there is time for the output queue to be processed
+            await self.send_lifespan("startup")
+        except TestClientError:
+            # Pass these through directly, so that test clients can assert on them
+            raise
+        except:  # noqa
+            # Any other exception is (almost) definitely passed through from the app under test
+            # So it means the lifespan protocol is not supported.
+            self._lifespan_task = None
```

Comment:
Maybe we could log the exception just in case. I prefer not to silence exceptions entirely without any feedback. Also, I guess it's important we notify the user that lifespan protocol is not supported

Reply:
Yeah that seems reasonable, I'll add it. This is the behaviour I'd expect many real ASGI server would have (to warn the user), so we should behave the same way.